finagle-core/src/test/scala/com/twitter/finagle/filter/RequestMeterFilterTest.scala

```diff
@@ -0,0 +1,84 @@
+package com.twitter.finagle.filter
+
+import java.util.concurrent.RejectedExecutionException
+
+import com.twitter.concurrent.AsyncMeter
+import com.twitter.conversions.time._
+import com.twitter.finagle.{Failure, Service}
+import com.twitter.util._
+import org.junit.runner.RunWith
+import org.scalatest.FunSuite
+import org.scalatest.junit.JUnitRunner
+import org.scalatest.mock.MockitoSugar
+import org.mockito.Mockito.when
+
+@RunWith(classOf[JUnitRunner])
+class RequestMeterFilterTest extends FunSuite with MockitoSugar {
+
+  val echoSvc = new Service[Int, Int] {
+    def apply(req: Int) = Future(req)
+  }
+
+  test("return service execution after getting a permit") {
+    val timer = new MockTimer
+    Time.withCurrentTimeFrozen { ctl =>
+      val meter = AsyncMeter.newMeter(1, 1 second, 1)(timer)
+      val svc = new RequestMeterFilter(meter) andThen echoSvc
```

Comment:
`.andThen(echoSvc)` here and elsewhere (infix style is only for symbolic operators)